cardano-db-sync/src/Cardano/DbSync/Era/Universal/Insert/Grouped.hs

```diff
@@ -85,100 +90,148 @@ instance Semigroup BlockGroupedData where
+    -- Individual process ID-based updates
+    unless (null idBasedUpdates) $
+      mapM_ DB.updateListTxOutConsumedByTxId idUpdateChunks
+    -- Log failures
+    mapM_ (liftIO . logWarning tracer . ("Failed to find output for " <>) . Text.pack . show) failedInputs
+
+  -- Process metadata and mint chunks
+  mapM_ (DB.insertBulkTxMetadata removeJsonbFromSchema) txMetadataChunks
+  mapM_ DB.insertBulkMaTxMint txMintChunks
+
+  pure $ makeMinId syncEnv txInIds txOutIds maTxOutIds
   where
     tracer = getTrace syncEnv
-    txOutTableType = getTxOutVariantType syncEnv
-
-    makeMinId :: [DB.TxInId] -> [DB.TxOutIdW] -> [DB.MaTxOutIdW] -> DB.MinIdsWrapper
-    makeMinId txInIds txOutIds maTxOutIds =
-      case txOutTableType of
-        DB.TxOutVariantCore -> do
-          DB.CMinIdsWrapper $
-            DB.MinIds
-              { minTxInId = listToMaybe txInIds
-              , minTxOutId = listToMaybe $ DB.convertTxOutIdCore txOutIds
-              , minMaTxOutId = listToMaybe $ DB.convertMaTxOutIdCore maTxOutIds
-              }
-        DB.TxOutVariantAddress ->
-          DB.VMinIdsWrapper $
-            DB.MinIds
-              { minTxInId = listToMaybe txInIds
-              , minTxOutId = listToMaybe $ DB.convertTxOutIdVariant txOutIds
-              , minMaTxOutId = listToMaybe $ DB.convertMaTxOutIdVariant maTxOutIds
-              }
+    txOutVariantType = getTxOutVariantType syncEnv
+    removeJsonbFromSchema = ioRemoveJsonbFromSchema $ soptInsertOptions $ envOptions syncEnv
+
+-- | Parallel implementation with single connection coordination
+insertBlockGroupedData ::
+  MonadIO m =>
+  SyncEnv ->
+  BlockGroupedData ->
+  DB.DbAction m DB.MinIdsWrapper
+insertBlockGroupedData syncEnv grouped = do
+  disInOut <- liftIO $ getDisableInOutState syncEnv
+
+  -- Parallel preparation of independent data
+  (preparedTxIn, preparedMetadata, preparedMint, txOutChunks) <- liftIO $ do
+    a1 <- async $ pure $ prepareTxInProcessing syncEnv grouped
+    a2 <- async $ pure $ prepareMetadataProcessing syncEnv grouped
+    a3 <- async $ pure $ prepareMintProcessing syncEnv grouped
+    a4 <- async $ pure $ chunksOf maxBulkSize $ etoTxOut . fst <$> groupedTxOut grouped
+
+    r1 <- wait a1
+    r2 <- wait a2
+    r3 <- wait a3
+    r4 <- wait a4
+    pure (r1, r2, r3, r4)
+
+  -- Sequential TxOut processing (generates required IDs)
+  txOutIds <- concat <$> mapM (DB.insertBulkTxOut disInOut) txOutChunks
+
+  -- Execute independent operations (TxIn, Metadata, Mint) in parallel
+  txInIds <- executePreparedTxIn preparedTxIn
+
+  -- TxOut-dependent operations (MaTxOut + UTxO consumption)
+  maTxOutIds <- processMaTxOuts syncEnv txOutIds grouped
+
+  -- Execute remaining independent operations in parallel with pools
+  liftIO $ do
+    a1 <- async $ DB.runPoolDbAction (envDbEnv syncEnv) (executePreparedMetadata preparedMetadata)
+    a2 <- async $ DB.runPoolDbAction (envDbEnv syncEnv) (executePreparedMint preparedMint)
```

Comment:
Not that using a separate connection from a pool, means these inserts no longer happen in the same open transaction as the rest of the data of the block. This means that we could have cases where half the block is inserted